modules/programs/tarsnap.nix

```diff
@@ -0,0 +1,300 @@
+          higher value is recommended since creating a checkpoint in an archive
+          can take a few seconds and several hundred kB of bandwidth.
+        '';
+      };
+
+      diskPause = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = ''
+          Pause for X ms between storing archive entries and after every 64 kB
+          of file data. This will slow down tarsnap and thereby reduce its
+          impact on other applications. For archiving files which are stored on
+          an ATA disk and are not in the operating system disk cache, a value of
+          diskPause = 10 will approximately double the time taken.
+        '';
+      };
+
+      exclude = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Do not process files or directories that match the specified pattern.
+          Note that exclusions take precedence over patterns or filenames
+          specified on the command line.
+        '';
+      };
+
+      forceResources = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Force the decryption of a passphrase-encrypted key file to proceed
+          even if it is anticipated to require an excessive amount of memory or
+          CPU time
+        '';
+      };
+
+      humanizeNumbers = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Use SI prefixes to make numbers printed by --print-stats and SIGINFO
+          more readable.
+        '';
+      };
+
+      include = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Process only files or directories that match the specified pattern.
+          Note that exclusions specified with exclude take precedence over
+          inclusions. If no inclusions are explicitly specified, all entries are
+          processed by default. The --include option is especially useful when
+          filtering archives. For example, the command <command>tarsnap -c -f
+          foo-backup --include='*foo*' @@all-backup</command> creates a new
+          archive foo-backup containing only the entries from all-backup
+          containing the string Sq foo.
+        '';
+      };
+
+      insaneFilesystems = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Allow descent into synthetic filesystems such as procfs. Normally
+          archiving of such filesystems is a silly thing to do, hence the name
+          of the option.
+        '';
+      };
+
+      isoDates = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Print file and directory dates as yyyy-mm-dd hh:mm:ss.
+        '';
+      };
+
+      keyfile = mkOption {
+        type = types.nullOr types.str;
```

Comment:
Set type to `types.nullOr types.path` and add `apply = toString;` to reduce risk of having the key file copied to the Nix store.